specs/core/1_shard-data-chains.md

```diff
@@ -17,595 +17,380 @@
 | `SHARD_HEADER_SIZE` | `2**9` (= 512) |
 | `SHARD_BLOCK_SIZE_TARGET` | `2**14` (= 16,384) |
-| `SHARD_BLOCK_SIZE_LIMIT` | `2**16` (= 65,536) |
+| `MAX_SHARD_BLOCK_SIZE` | `2**16` (= 65,536) |
 
 ### Initial values
 
 | Name | Value |
 | - | - |
-| `PHASE_1_FORK_EPOCH` | **TBD** |
-| `PHASE_1_FORK_SLOT` | **TBD** |
+| `SHARD_GENESIS_EPOCH` | **TBD** |
 
 ### Time parameters
 
 | Name | Value | Unit | Duration |
 | - | - | :-: | :-: |
-| `CROSSLINK_LOOKBACK` | `2**0` (= 1) | epochs | 6.4 minutes |
+| `SHARD_SLOTS_PER_EPOCH` | `2**7` (= 128) | shard slots | 6.4 minutes |
 | `EPOCHS_PER_SHARD_PERIOD` | `2**8` (= 256) | epochs | ~27 hours |
 
 ### State list lengths
 
-| Name | Value | Unit |
-| - | - | :-: |
-| `HISTORY_ACCUMULATOR_VECTOR` | `2**6` (= 64) | state tree maximum depth |
+| Name | Value |
+| - | - |
+| `HISTORY_ACCUMULATOR_DEPTH` | `2**6` (= 64) |
 
 ### Rewards and penalties
 
 | Name | Value |
 | - | - |
-| `BASEFEE_ADJUSTMENT_FACTOR` | `2**3` (= 8) |
-| `REWARD_COEFFICIENT_BASE` | `2**20` ( = 1,048,576) |
+| `BLOCK_BODY_PRICE_QUOTIENT` | `2**3` (= 8) |
 
 ### Signature domain types
 
-The following types are defined, mapping into `DomainType` (little endian):
-
 | Name | Value |
 | - | - |
 | `DOMAIN_SHARD_PROPOSER` | `128` |
 | `DOMAIN_SHARD_ATTESTER` | `129` |
 
-### TODO PLACEHOLDER
-
-| Name | Value |
-| - | - |
-| `PLACEHOLDER` | `2**3` |
-
-## Data structures
-
-_Note: the shard block header structure is carefully designed so that all of the values have the same depth in a hash tree implementation, so `hash_tree_root(SSZ_partial(x)) == hash_tree_root(x)` (using the "left-to-right leaves" scheme [here](https://github.com/ethereum/eth2.0-specs/issues/1303)), which allows shard block headers to look like an SSZ object when in the crosslink structure. This is done by balancing it so that 7 or 8 items are on the left side (the "core") and two 96-byte (ie. 3*2 = 6 chunk) items are on the right side. Change with care._
-
-### `ShardBlockHeader`
-
-```python
-class ShardBlockHeader(Container):
-    core: ShardBlockCore
-    signatures: ShardBlockSignatures
-```
+## Containers
 
 ### `ShardBlock`
 
 ```python
-class ShardBlock(Container):
-    core: ExtendedShardBlockCore
-    signatures: ShardBlockSignatures
-```
-
-### `ShardBlockSignatures`
-
-```python
-class ShardBlockSignatures(Container):
-    attestation_signature: BLSSignature
-    proposer_signature: BLSSignature
-```
-
-### `ShardBlockCore`
-
-```python
-class ShardBlockCore(Container):
+class ShardBlock(FlatContainer):
+    shard: Shard
     slot: ShardSlot
-    beacon_chain_root: Hash
+    beacon_block_root: Hash
     parent_root: Hash
-    data_root: Hash
     state_root: Hash
-    total_bytes: uint64
-    attester_bitfield: Bitvector[TARGET_PERSISTENT_COMMITTEE_SIZE * 2]
+    body: List[byte, MAX_SHARD_BLOCK_SIZE - SHARD_HEADER_SIZE]
+    block_size_sum: uint64
+    aggregation_bits: Bitvector[2 * MAX_PERIOD_COMMITTEE_SIZE]
```

Comment:
I'm a bit torn on the `ShardCheckpoint` and bits/signature construction currently. 

1. `ShardCheckpoint` isn't actually a checkpoint so we should consider renaming. Maybe `ShardAttestationData`
2. We have no direct datastructure mapped to what would be sent on the network. For example, in Phase 0, an `Attestation` is nice because it is the precise construction of the datastructure sent on the network. I'm considering defining a `ShardAttestation` for this purpose. Considering defining here _even if_ it is not actually used as a consensus object in the state transition. When we define the phase 1 network spec, if it isn't used in consensus, we can pull it out then.

Reply:
Agreed on renaming to `ShardAttestationData`. Defining `ShardAttestation` in the networking and/or validator doc makes sense to me.

Reply:
yeah yeah. Validator doc might be the right place